pkgs/os-specific/linux/kernel/manual-config.nix

```diff
@@ -220,6 +220,14 @@ let
         # from drivers/ in the future; it adds 50M to keep all of its
         # headers on 3.10 though.
 
+        # Keep generated certificates to sign out of tree modules
+        mkdir -p $dev/lib/modules/${modDirVersion}/build/certs
+        for f in certs/signing_key.pem certs/signing_key.x509; do
+          if [ -f "$buildRoot/$f" ]; then
+            cp $buildRoot/$f $dev/lib/modules/${modDirVersion}/build/certs
+          fi
+        done
+
```

Comment:
With the private signing key in the store, wouldn't this allow anyone on the system to sign arbitrary modules?

Reply:
I thought it wouldn't be as much of a problem as the certs are generated on every kernel build since we don't supply one by default.

I also assumed that derivations in /nix/store wouldn't be writable by other things (if that's compromised I don't think anything I could do would be much better).

Reply:
I'm not sure I understand? As long as the key is in the store, you can just take it and sign an arbitrary module with it, without ever needing to touch anything else in the nix store, right?

Reply:
For my specific use case for this PR, it's to modprobe out of tree modules doing the boot phase. The only modules in the initrd are the result of a nix derivation so I'd expect that nothing would be compromised there.

I'm not sure if you can `modprobe` modules that aren't in the nix store, but if you can, then yeah, that would be a problem.

Reply:
I think you can modprobe a module from anywhere if it's binary-compatible with your kernel.

Let's imagine the scenario: I somehow gained access to the `insmod` command (it loads a kernel module from a file and is a low-level interface to module loading facilities) and able to pass it any path (incorrect SUID configuration on a vulnerable program?). I could craft a kernel module for the same kernel you use (and I could make a reasonable guess by comparing the hashes, if you use a standard Nixpkgs kernel) and then inject it into the kernel.

If the private keys for signing kernel modules are in the /nix/store, I could try to exfiltrate them using the hashes to find the kernel directory (or just by using /run/booted-system?) and sign the module with them. Bam, I have kernel-level access.

Reply:
One use-case I could imagine would be the builder nix store being kept secret (e.g. separate, access-restriced machine). 

Without access to that, the security would hold on the target machine.